src/components/search-result-list/SearchResultList.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from "react";
+import { formatFilters } from "../../apps/search-result/helpers";
+import { FilterItemTerm } from "../../apps/search-result/types";
+import {
+  FacetField,
+  useSearchFacetQuery
+} from "../../core/dbc-gateway/generated/graphql";
+import { useCampaignMatchPOST } from "../../core/dpl-cms/dpl-cms";
+import {
+  CampaignMatchPOST200,
+  CampaignMatchPOSTBodyItem
+} from "../../core/dpl-cms/model";
+import { getCoverTint, isObjectEmpty } from "../../core/utils/helpers/general";
+import { Work } from "../../core/utils/types/entities";
+import Campaign from "../campaign/Campaign";
+import SearchResultListItem from "./search-result-list-item/search-result-list-item";
+
+export interface SearchResultListProps {
+  resultItems: Work[];
+  filters: { [key: string]: { [key: string]: FilterItemTerm } };
+  q: string;
+}
+
+export const allFacetFields = [
+  FacetField.MainLanguages,
+  FacetField.AccessTypes,
+  FacetField.ChildrenOrAdults,
+  FacetField.Creators,
+  FacetField.FictionNonfiction,
+  FacetField.FictionalCharacter,
+  FacetField.GenreAndForm,
+  FacetField.MaterialTypes,
+  FacetField.Subjects,
+  FacetField.WorkTypes
+];
+
+const SearchResultList: React.FC<SearchResultListProps> = ({
+  resultItems,
+  filters,
+  q
+}) => {
+  const { mutate } = useCampaignMatchPOST();
+  const [campaignData, setCampaignData] = useState<CampaignMatchPOST200 | null>(
+    null
+  );
+  const { data } = useSearchFacetQuery({
+    q: { all: q },
+    facets: allFacetFields,
+    facetLimit: 10,
+    ...(isObjectEmpty(filters)
+      ? {}
+      : { filters: { ...formatFilters(filters) } })
+  });
```

Comment:
Please reconsider where in the stack facets are retrieved and whether it can be pushed up.

As far as I can tell this implies that we issue another facet query from the search result list to get the facets to use for campaign fetching. This is now done twice: Here and in `FacetBrowserModal.tsx`.

As I see it is would be preferable to do it once in `search-result.tsx` and then passed down to subcomponents.

Reply:
Yeah that's what I thought too, so I moved it up, but then @spaceo told me that the call is cached (which I think is very cool), so the second query just uses the cached data.

Reply:
We disagree here.

I do not think we should rely on cross-component query caching unless we have to. With both react-query and Redux it means we have multiple state managers which does not sound like a good idea. With components being almost adjacent in the tree there is little work in passing the facets down and on the other hand we risk the two queries moving out of sync and thus accidentally making the cache unusable.

If we wanted to render the facet browser on its own in Storybook it would be useful but that is not the case at the moment.

Oh well.  I said my piece. We leave it as is.